### PR TITLE
feat: simplify node start

### PR DESCRIPTION
**Pull request by redhdx**
### Description

Simplify op-geth start.

### Rationale

Currently, starting op-geth requires manual setting of many parameters. In order to facilitate users to start nodes, some default configuration added for opBNB mainnet and testnet.

### Example

Currently, there are two network configurations, opBNB mainnet and testnet by default. You can select the corresponding network configuration by setting `--opBNBMainnet` or `--opBNBTestnet`.  You also need to set some config manually:
* `--authrpc.jwtsecret=./jwt.txt`
* `--rollup.sequencerhttp=L2_RPC`
* `--datadir=./datadir`

You can start the node in the following way. If you are starting a new node, you do not need to init genesis.json. If you just restart the node and have already inited genesis.json, you can still use this command. You do not need to set `--datadir`. The op-geth will provide a default datadir. You can see the data path in the startup log. like this:
```
 database=/.../Ethereum/opBNBTestnet/geth/chaindata
```
```
database=/.../Ethereum/opBNBMainnet/geth/chaindata
```
However, if you have set datadir before and started node, you must still set it and keep it consistent. It is recommended to set datadir manually.

For example:
```
 ./geth --opBNBMainnet --authrpc.jwtsecret=./jwt.txt --datadir=./datadir --rollup.sequencerhttp=L2_RPC
```
```
 ./geth --opBNBTestnet --authrpc.jwtsecret=./jwt.txt --datadir=./datadir --rollup.sequencerhttp=L2_RPC
```

### Changes

Notable changes:
* add some default settings for `opBNB`

### CODE REVIEW

File: core/txpool/legacypool/legacypool.go [hunk 1]
```diff
@@ -171,6 +171,24 @@ var DefaultConfig = Config{
 	ReannounceTime: 10 * 365 * 24 * time.Hour,
 }
 
+// DefaultOpBNBConfig contains the default opBNB configurations for the transaction pool.
+var DefaultOpBNBConfig = Config{
```
Comment (owen-reorg) [inline core/txpool/legacypool/legacypool.go:175]: I think we can keep it same with DefaultConfig

File: miner/miner.go [hunk 2]
```diff
@@ -102,6 +102,19 @@ var DefaultConfig = Config{
...
+// DefaultOpBNBConfig contains default opBNB settings for miner.
+var DefaultOpBNBConfig = Config{
+	GasCeil:  150000000,
+	GasPrice: big.NewInt(params.GWei),
+
+	// The default recommit time is chosen as two seconds since
+	// consensus-layer usually will wait a half slot of time(6s)
+	// for payload generation. It should be enough for Geth to
+	// run 3 rounds.
+	Recommit:          2 * time.Second,
```
Comment (owen-reorg) [inline miner/miner.go:114]: seems the context does not match opbnb

File: miner/miner.go [hunk 3]
```diff
@@ -102,6 +102,19 @@ var DefaultConfig = Config{
 	NewPayloadTimeout: 2 * time.Second,
 }
 
+// DefaultOpBNBConfig contains default opBNB settings for miner.
+var DefaultOpBNBConfig = Config{
+	GasCeil:  150000000,
```
Comment (owen-reorg) [inline miner/miner.go:107]: what's this change for?

File: cmd/utils/flags.go [hunk 4]
```diff
@@ -1826,6 +1828,14 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 	log.Debug("Sanitizing Go's GC trigger", "percent", int(gogc))
 	godebug.SetGCPercent(int(gogc))
 
+	// set gc mode default value is archive for opBNB
+	if !ctx.IsSet(GCModeFlag.Name) && (ctx.Bool(OpBNBMainnetFlag.Name) || ctx.Bool(OpBNBTestnetFlag.Name)) {
+		err = ctx.Set(GCModeFlag.Name, "archive")
```
[the remark this answers is not in the document]
Comment (redhdx) [reply cmd/utils/flags.go]: Resovled. The default value is `full`, so just remove this change. According to proposer submit block interval, we set `TrieCommitInterval` value is 3600 in mainnet and 240 in testent. Please help confirm it.

File: core/txpool/legacypool/legacypool.go (hunk 1 above)
Comment (redhdx) [reply core/txpool/legacypool/legacypool.go]: Agree, user can set the txpool config manually. Resovled. Please help confirm it again.

File: miner/miner.go (hunk 2 above)
Comment (redhdx) [reply miner/miner.go]: This is based on the previous startup document. In fact, this configuration is useless. This is the previous block production configuration of Ethereum. For us, we only have the corresponding parameters for the master sequencer to produce blocks and control the block production by other conditions. 
Remove it. Please help confirm it again.

File: miner/miner.go (hunk 3 above)
Comment (redhdx) [reply miner/miner.go]: Same as above.

**Review by owen-reorg — APPROVED** [review]

File: cmd/geth/main.go [hunk 5]
```diff
@@ -348,10 +348,11 @@ func prepare(ctx *cli.Context) {
...
 		}
-	} else if ctx.String(utils.SyncModeFlag.Name) != "light" && !ctx.IsSet(utils.CacheFlag.Name) && ctx.IsSet(utils.OpBNBMainnetFlag.Name) {
-		// we're really on opBNB mainnet. Bump that cache up
-		log.Info("Bumping default cache on mainnet", "provided", ctx.Int(utils.CacheFlag.Name), "updated", 4096, "network", ctx.String(utils.OpBNBMainnetFlag.Name))
-		ctx.Set(utils.CacheFlag.Name, strconv.Itoa(4096))
+	} else if ctx.String(utils.SyncModeFlag.Name) != "light" && !ctx.IsSet(utils.CacheFlag.Name) &&
+		(ctx.IsSet(utils.OpBNBMainnetFlag.Name) || ctx.IsSet(utils.OpBNBTestnetFlag.Name)) {
+		// we're really on opBNB network. Bump that cache up
+		log.Info("Bumping default cache on opBNB", "provided", ctx.Int(utils.CacheFlag.Name), "updated", 32000)
+		ctx.Set(utils.CacheFlag.Name, strconv.Itoa(32000))
```
Comment (welkin22) [inline cmd/geth/main.go:355]: 32000 seems too high, currently our mainnet node appears to be at 22000, can you confirm again?

**Review by welkin22 — COMMENTED** [review]
LGTM, except for one question

File: cmd/geth/main.go (hunk 5 above)
Comment (redhdx) [reply cmd/geth/main.go]: Yes, our doc is still 32000, but i think 22000 is more reasonable. Updated, pls confirm it.

**Review by welkin22 — APPROVED** [review]
LGTM

**Review by owen-reorg — APPROVED** [review]

**Pull request merged**